wolfcrypt/src/evp.c

```diff
@@ -6878,6 +6878,446 @@ int wolfSSL_EVP_get_hashinfo(const WOLFSSL_EVP_MD* evp,
+
+    WOLFSSL_ENTER("wolfSSL_EVP_EncodeFinal");
+
+    if( outl == NULL)
+        return;
+
+    if( ctx == NULL || out == NULL  ) {
+        *outl = 0;
+        return;
+    }
+    /* process remaining data in ctx */
+    outsz = BASE64_ENCODED_BLOCK_SIZE + 1;   /* 64 byte and one for LF*/
+    res = Base64_Encode( ctx->data, ctx->remaining ,out, &outsz);
+    if( res == 0)
+        *outl = outsz;
+    else
+        *outl = 0;
+
+    ctx->remaining = 0;
+    XMEMSET( ctx->data,0,sizeof(ctx->data));
+
+    return;
+}
+#endif /* WOLFSSL_BASE64_ENCODE */
+#if defined(WOLFSSL_BASE64_DECODE)
+/* borrowed from coding.c */
+static WC_INLINE int Base64_SkipNewline(const byte* in, word32* inLen,
+             word32* outJ)
+{
+    word32 len = *inLen;
+    word32 j = *outJ;
+    if (len && (in[j] == ' ' || in[j] == '\r' || in[j] == '\n')) {
+        byte endLine = in[j++];
+        len--;
+        while (len && endLine == ' ') {   /* allow trailing whitespace */
+            endLine = in[j++];
+            len--;
+        }
+        if (endLine == '\r') {
+            if (len) {
+                endLine = in[j++];
+                len--;
+            }
+        }
+        if (endLine != '\n') {
+            WOLFSSL_MSG("Bad end of line in Base64 Decode");
+            return ASN_INPUT_E;
+        }
+    }
+    if (!len) {
+        return BUFFER_E;
+    }
+    *inLen = len;
+    *outJ = j;
+    return 0;
+}
+/*  wolfSSL_EVP_DecodeInit initializes specified WOLFSSL_EVP_ENCODE_CTX struct
+ *  for subsequent wolfSSL_EVP_DecodeUpdate.
+ */
+void wolfSSL_EVP_DecodeInit(WOLFSSL_EVP_ENCODE_CTX* ctx)
+{
+    WOLFSSL_ENTER("wolfSSL_EVP_DecodeInit");
+    /* clean up ctx */
+    if(ctx != NULL) {
+        ctx->remaining = 0;
+        XMEMSET(ctx->data,0, sizeof(ctx->data));
+    }
+}
+/*  wolfSSL_EVP_DecodeUpdate encodes the input data in 4-byte units
+ *  and outputs it to out. If less than 4 bytes of data remain, save it in
+ *  ctx. The data given in the subsequent wolfSSL_EVP_DecodeUpdate
+ *  is combined with the data stored in CTX and used for decoding.
+ *  Returns 1 or 0 on success, -1 on error. Return value 0 indicates that
+ *  clients should call wolfSSL_EVP_DecodeFinal as next call.
+ */
+int  wolfSSL_EVP_DecodeUpdate(WOLFSSL_EVP_ENCODE_CTX* ctx,
+            unsigned char*out, int *outl, const unsigned char*in, int inl)
+{
+    word32 outsz = 0;
+    word32 j = 0;
+    word32 inLen;
+    int    res;
+    int    pad = 0;
+    int    i;
+    char es;
+
+    WOLFSSL_ENTER("wolfSSL_EVP_DecodeUpdate");
+
+    if (outl == NULL)
+        return -1;
+
+    if (ctx == NULL || out == NULL || in == NULL ) {
+        *outl = 0;
+        return -1;
+    }
+
+    if (inl == 0) {
+        *outl = 0;
+        return 1;
+    }
+
+    inLen = inl;
+    *outl = 0;
+
+    /* if the remaining data exist in the ctx, add input data to them to create
+    a block(4bytes) for decoding*/
+    if( ctx->remaining > 0 && inl > 0) {
+        int cpysz;
+        unsigned char e0;
```

Comment:
Can we change the variable name here?  Oftentimes e0 refers to exponentiation which could cause confusion.

Reply:
Renamed variable "e0" to "c".